Classes/Notifications/NotificationClient.swift

```diff
@@ -59,13 +59,25 @@ final class NotificationClient {
         width: CGFloat,
         completion: @escaping (Result<([NotificationViewModel], Int?)>) -> Void
         ) {
-        githubClient.client.send(V3NotificationRequest(all: all, page: page)) { result in
-            switch result {
-            case .success(let response):
-                let viewModels = CreateViewModels(containerWidth: width, v3notifications: response.data)
-                self.fetchStates(for: viewModels, page: response.next, completion: completion)
-            case .failure(let error):
-                completion(.error(error))
+        if let repo = repo {
+            githubClient.client.send(V3RepositoryNotificationRequest(all: all, owner: repo.owner, repo: repo.name)) { result in
+                switch result {
```

Comment:
I want to not repeat myself with the completionHandler, but as it relies on the generic type of the Request, I couldn't find a way. Any ideas?

Reply:
I would wrap this into a function: 
```
switch result {
      case .success(let response):
           let viewModels = CreateViewModels(containerWidth: width, v3notifications: response.data)
           self.fetchStates(for: viewModels, page: response.next, completion: completion)
      case .failure(let error):
           completion(.error(error))
}
```

What do you think?

Reply:
That's exactly what I wanted to do, but as the result is generic over `T` (from the `send()` function), the compiler won't allow it.

Reply:
Probably fine for now, not too offended